app/models/state_changers.rb

```diff
@@ -201,47 +191,48 @@ def contents_for(target_state)
       return nil unless FILTER_FAILS_ON.include?(target_state)
 
       # determine list of well locations requiring the state change
-      well_locations_filtered = labware.wells.reject { |w| w.state == 'failed' }.map(&:location)
+      well_locations_filtered = v2_labware.wells.reject { |w| w.state == 'failed' }.map(&:location)
```

Comment:
Maybe the 'failed' here should be a constant, like FILTER_FAILS_ON.

Also, I think I wondered before why on line 191 'failed' is in FILTER_FAILS_ON (so target_state can be 'failed', 'cancelled' or 'qc_complete') but then we filter out the 'failed' wells on line 194 (but not cancelled or qc_complete). Is that deliberate?
Does anyone understand that?